pkg/etcd/client.go

```diff
@@ -168,6 +172,41 @@ func (c *Client) Watch(ctx context.Context, key string, opts ...clientv3.OpOptio
 	return c.cli.Watch(ctx, key, opts...)
 }
 
+// WatchWithChan maintains a watchCh and send all msg from the watchCh to outCh
+func (c *Client) WatchWithChan(ctx context.Context, outCh chan clientv3.WatchResponse, key string, revision int64) {
```

Comment:
Can you moving calling `RequestProgress` into this function?

Reply:
ok